tsdb/db.go

```diff
@@ -574,13 +589,16 @@ func validateOpts(opts *Options, rngs []int64) (*Options, []int64) {
 	return opts, rngs
 }
 
-func open(dir string, l log.Logger, r prometheus.Registerer, opts *Options, rngs []int64) (_ *DB, returnedErr error) {
+func open(dir string, l log.Logger, r prometheus.Registerer, opts *Options, rngs []int64, stats *DBStats) (_ *DB, returnedErr error) {
```

Comment:
Why not just returning one?

Reply:
The whole problem in the first place is that the `open` function triggers the WAL replay, and we have to wait for that to finish until we get the return. So if we return the stats object, that means we can't get any of the WAL replay progress until it's actually completed, which defeats the whole point.

Reply:
Ah you are right, thanks. Alternatively we could return unready DB, but not worth it (: